### PR TITLE
Fix Pull Request comment filename word breaks

### LINKED ISSUE #16248: Incorrect wrapping on PR comments with long paths
- Gitea version (or commit ref): 1.14.3
- Git version: 2.25.1
- Operating system: Ubuntu Linux 20.04.2
- Built Gitea myself, running from systemd (my own service script) on an AWS x86_64 box
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1
  - [ ] No

## Description

Comments on files with long paths will not wrap the file name header on a PR if there's no easily-available word boundary to wrap on. This causes the "Outdated" badge and/or "Show Resolved"/"Hide Resolved" links to either overlap other UI elements or be completely obscured, rendering them un-clickable. Note: this is not a contrived example, we ran into this on our project.

Example of a long path with available word boundaries: https://try.gitea.io/parnic-sks/long-paths-test/pulls/1
Example of a long path that doesn't currently wrap: https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1

## Screenshots

Working (valid word boundaries):
image

Not working (no valid word boundaries):
image

## Potential fix

Setting `word-break: break-word;` CSS on one of the `segment` bits used for PR comment filename headers is one possible way to fix the issue:
image

### PR DESCRIPTION
**Pull request by parnic**
This fixes word wrapping on the filename for a comment on a PR. A previous commit fixed this problem in Firefox, but not Chrome.

It also fixes the Outdated badge wrapping mid-word in the same situation. (https://github.com/go-gitea/gitea/issues/16248#issuecomment-869703289)

Fixes #16248

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Even for Firefox, the bug still has existed, so I believe the old fix was incorrect.

I have a feeling that we should remove all the `word-break-*` classes, and just add a `word-break-all` to `<a href="{{(index $comms 0).CodeCommentURL}}" class="file-comment ml-3">{{$filename}}</a>`

I have tested it on https://try.gitea.io/wxiaoguang/test/pulls/9 

### Before

image

### Remove all word-break and just add one to `<a>`

image

**Comment by parnic** [comment]
@wxiaoguang I pushed a similar change to what you're recommending. It does work when applied to https://try.gitea.io/parnic-sks/long-paths-test-2/pulls/1, but if you'd prefer it be `break-all` instead of `break-word`, I can do that.

**Comment by silverwind** [comment]
FYI, `word-break: break-word` is deprecated as per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#values). I don't know the intrinsic details, but I think `overflow-wrap` is supposed to replace it.

**Comment by parnic** [comment]
> FYI, `word-break: break-word` is deprecated as per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#values). I don't know the intrinsic details, but I think `overflow-wrap` is supposed to replace it.

After testing, it looks like `overflow-wrap: anywhere` behaves the same. I've added that to the class used here.

File: web_src/less/helpers.less [hunk 1]
```diff
@@ -27,6 +27,8 @@
 
 .word-break {
   word-wrap: break-word !important;
+  word-break: break-word;
```
Comment (silverwind) [inline web_src/less/helpers.less]:
```suggestion
  word-break: break-word; /* compat: Safari */
```

**Review by 6543 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**